src/NuGet.Core/NuGet.DependencyResolver.Core/Remote/RemoteDependencyWalker.cs

```diff
@@ -552,4 +554,100 @@ public int GetHashCode(LibraryDependency obj)
             }
         }
     }
+
+    internal struct LightweightList<T>
+    {
+        private const int Fields = 4;
```

Comment:
Why 4? 

I'm curious, because I've run queries to get the avg number of dependencies for package son nuget.org and the answer is 3.1.

```kql
NiPackageManifests
| join kind=inner NiPackageVersions on Identity
| where IsListed
| extend parsedGroups = parse_json(DependencyGroups)
| mv-expand parsedGroups
| extend DependencyCount = iff(isempty(parsedGroups['Packages']), 0, array_length(parsedGroups['Packages']))
| summarize avg(DependencyCount)
```

Reply:
It was more or less a guess. I chose 4 because it's the default size for `List<T>` and because the existing code initialized the list to 1. Looks like 4 is in the ballpark of being a good heuristic. I'd be curious if we can do better though. For instance, if the 90th percentile is 4.8 (or something), making this 5 would be better.

Reply:
> I chose 4 because it's the default size for List<T>

The initial capacity when something gets added for the sleuths :D